contracts/drafts/ERC777/ERC777.sol

```diff
@@ -144,7 +144,7 @@ contract ERC777 is IERC777, ERC20Detailed {
      * @param operator address to be authorized as operator
      */
     function authorizeOperator(address operator) external {
-        require(msg.sender != operator);
+        require(msg.sender != operator, "ERC777: operator authorization to self");
```

Comment:
What do you think about phrasing this as: `authorizing self as operator`? Not sure if we used this style anywhere else, but at least in this case I think it's a lot clearer.

Reply:
Yeah that was also my first take, but I don't think any of the other messages use gerund. If you think it is indeed easier to understand I'll change it.